toolchain/base/value_store.h

```diff
@@ -72,9 +77,30 @@ struct FloatId : public IdBase, public Printable<FloatId> {
 };
 constexpr FloatId FloatId::Invalid(FloatId::InvalidIndex);
 
+// DenseMapInfo for llvm::APFloat, for use in the canonical float value store.
+// TODO: Remove this once our new hash table is available.
+struct APFloatDenseMapInfo {
```

Comment:
We can't specialize `llvm::DenseMapInfo<llvm::APFloat>` ourselves because we don't own either of those types. I didn't find that implementation in `LLVMContextImpl.h` when searching -- I wonder why it's *there* of all places; that doesn't make any sense at all. But we can't use that anyway because it's not in a public header file.

I looked at `Bogus` but from a brief inspection it looked like `APFloat` would assert if you tried to copy it in that state. But maybe I was wrong -- if LLVM itself is doing this then I assume it's fine. Switching to using that.